### PR TITLE
CORE-793: Address Memory Leak Issues

### PR DESCRIPTION
**Pull request by EBGToo**
These leaks were found when displaying and submitting XRP transfers but some of them applied outside of XRP.

I'll run some BTC, etc tests and perhaps add to this PR.

### CODE REVIEW

**Comment by EBGToo** [comment]
@carlcherry If you didn't look at it carefully, can you confirm the 'ripple' changes in detail?

File: ripple/BRRippleTransaction.c [hunk 1]
```diff
@@ -318,10 +318,11 @@ rippleTransactionSerializeAndSign(BRRippleTransaction transaction, BRKey * priva
     // Serialize and sign the tx bytes
     BRRippleSerializedTransaction serializedBytes = rippleTransactionSerializeImpl (transaction, 0, 0);
     BRRippleSignature sig = signBytes(privateKey, serializedBytes->buffer, serializedBytes->size);
-    free(serializedBytes); // Free the serialized bytes since we have to reserialize with signature below
```
Comment (carlcherry) [inline ripple/BRRippleTransaction.c]: I guess I missed the fact that this structure had a member that was also allocated memory

Comment (carlcherry) [reply ripple/BRRippleTransaction.c]: I know this is not part of your change - but I do notice that down on line 335 we are dereferencing transaction->signedBytes without checking if the pointer is valid

File: ripple/BRRippleTransaction.c [hunk 2]
```diff
@@ -318,10 +318,11 @@ rippleTransactionSerializeAndSign(BRRippleTransaction transaction, BRKey * priva
...
 
     transaction->signedBytes = rippleTransactionSerializeImpl(transaction, sig->signature, sig->sig_length);
-
+    
+    rippleSignatureDelete(sig);
```
Comment (carlcherry) [inline ripple/BRRippleTransaction.c]: Yes - forgot to free this.

File: generic/BRGenericRipple.c [hunk 3]
```diff
@@ -212,12 +212,16 @@ genericRippleWalletCreateTransfer (BRGenericWalletRef wallet,
...
+    BRRippleUnitDrops drops = amount.u64[0];
 
-    return (BRGenericTransferRef) rippleTransferCreateNew (source,
-                                                           (BRRippleAddress) target,
-                                                           drops);
+    BRRippleTransfer transfer = rippleTransferCreateNew (source,
+                                                         (BRRippleAddress) target,
+                                                         drops);
+
+    rippleAddressFree(source);
```
Comment (carlcherry) [inline generic/BRGenericRipple.c]: Yes - this is needed.

File: ripple/BRRippleTransaction.c (hunk 1 above)
Comment (EBGToo) [reply ripple/BRRippleTransaction.c]: Yes, that needs to be fixed!

Comment (EBGToo) [reply ripple/BRRippleTransaction.c]: Like this `return (NULL == transaction->signedBytes ? 0 : transaction->signedBytes->size);`  Okay?

**Pull request merged**